service/worker/migration/activities.go

```diff
@@ -746,13 +748,23 @@ const (
 
 func (a *activities) VerifyReplicationTasks(ctx context.Context, request *verifyReplicationTasksRequest) (verifyReplicationTasksResponse, error) {
 	ctx = headers.SetCallerInfo(ctx, headers.NewPreemptableCallerInfo(request.Namespace))
-	remoteClient := a.clientFactory.NewRemoteAdminClientWithTimeout(
-		request.TargetClusterEndpoint,
-		admin.DefaultTimeout,
-		admin.DefaultLargeTimeout,
-	)
-
 	var response verifyReplicationTasksResponse
+	var remoteClient adminservice.AdminServiceClient
+	var err error
+
+	if len(request.TargetClusterName) > 0 {
+		remoteClient, err = a.clientBean.GetRemoteAdminClient(request.TargetClusterName)
+		if err != nil {
+			return response, err
+		}
+	} else {
+		remoteClient = a.clientFactory.NewRemoteAdminClientWithTimeout(
```

Comment:
for backward-compatible

Reply:
add a TODO to remove this branch after the next release?